ambry-server/src/integration-test/java/com.github.ambry.server/RouterServerTestFramework.java

```diff
@@ -363,17 +366,22 @@ private void continueChain(final OperationChain opChain) {
         opChain.latch.countDown();
         return;
       }
+      GetOptions options = GetOptions.None;
       switch (nextOp) {
         case PUT:
           startPutBlob(opChain);
           break;
+        case GET_INFO_DELETED_SUCCESS:
+          options = GetOptions.Include_Deleted_Blobs;
         case GET_INFO:
         case GET_INFO_DELETED:
-          startGetBlobInfo(nextOp.afterDelete, opChain);
+          startGetBlobInfo(options, nextOp.afterDelete && options.equals(GetOptions.None), opChain);
```

Comment:
could you instead change the flag from `nextOp.afterDelete` to `nextOp.checkDeleted`. Then, we wouldn't need to have check the GetOptions type here.